src/multi_vector_simulator/D2_model_constraints.py

```diff
@@ -75,35 +75,25 @@ def add_constraints(local_energy_system, dict_values, dict_model):
     - D2.test_add_constraints_net_zero_energy_requirement_is_true()
     - D2.test_add_constraints_net_zero_energy_requirement_is_false()
     """
-    count_added_constraints = 0
 
-    if dict_values[CONSTRAINTS][MINIMAL_RENEWABLE_FACTOR][VALUE] > 0:
-        # Add minimal renewable factor constraint
-        local_energy_system = constraint_minimal_renewable_share(
-            local_energy_system, dict_values, dict_model
-        )
-        count_added_constraints += 1
+    constraint_functions = {
+        MINIMAL_RENEWABLE_FACTOR: constraint_minimal_renewable_share,
+        MAXIMUM_EMISSIONS: constraint_maximum_emissions,
+        MINIMAL_DEGREE_OF_AUTONOMY: constraint_minimal_degree_of_autonomy,
+        NET_ZERO_ENERGY: constraint_net_zero_energy,
+    }
 
-    if dict_values[CONSTRAINTS][MAXIMUM_EMISSIONS][VALUE] is not None:
-        # Add maximum emissions constraint
-        local_energy_system = constraint_maximum_emissions(
-            local_energy_system, dict_values
-        )
-        count_added_constraints += 1
+    count_added_constraints = 0
 
-    if dict_values[CONSTRAINTS][MINIMAL_DEGREE_OF_AUTONOMY][VALUE] > 0:
-        # Add minimal renewable factor constraint
-        local_energy_system = constraint_minimal_degree_of_autonomy(
+    for constraint in dict_values[CONSTRAINTS]:
```

Comment:
If I read this correctly, we still only apply constraints if they are ie. `>0` or `!=None`, but you placed the check into the constraint function itself. That works, eventhough I like it more to have less if-loops in the `constraint_functions`.

Reply:
Is it increasing simulation time to provide the big `local_energy_system` to the `constraint_functions`, compared to skipping the functions if we check whether they need to be applied (previous code?)

Reply:
I would intuitively say that less constraints definitions should lead to quicker simulations, but I am not 100% sure

Reply:
Well, if it is not major I guess it is allright.